fix: check-api-key Makefile target blocks install for non-OpenAI providers

## Description
Running `make helm-install` fails when using a non-OpenAI provider (e.g. Anthropic, Ollama, Gemini):
```
Error: OPENAI_API_KEY environment variable is not set for OpenAI provider
Please set it with: export OPENAI_API_KEY=your-api-key
make: *** [check-api-key] Error 1
```

## Root cause
After digging into the Makefile, the `check-api-key` target is actually 
provider-aware and handles Anthropic, Gemini, Azure, and Ollama correctly. 

The real issue is line 117:
```makefile
KAGENT_DEFAULT_MODEL_PROVIDER ?= openAI
```
The variable defaults to `openAI` silently. If you don't know this variable 
exists, you hit the OpenAI key check with no obvious way out. 
`DEVELOPMENT.md` makes no mention of it.

## Steps to reproduce
1. Set `ANTHROPIC_API_KEY` but not `OPENAI_API_KEY`
2. Run `make helm-install` without setting `KAGENT_DEFAULT_MODEL_PROVIDER`
3. Install fails — the Anthropic key is ignored because the provider defaulted to OpenAI

## Expected behavior
`DEVELOPMENT.md` should document `KAGENT_DEFAULT_MODEL_PROVIDER` and its 
supported values so users know to set it when using a non-OpenAI provider.

## Proposed fix
- Document `KAGENT_DEFAULT_MODEL_PROVIDER` in `DEVELOPMENT.md` with supported values and an example
- Optionally add an inline comment on line 117 listing supported values

## Environment
- kagent version: v0.8.0-beta7
- OS: macOS
- Provider: Anthropic (`claude-sonnet-4-20250514`)
```

--- a/DEVELOPMENT.md
+++ b/DEVELOPMENT.md
@@ -72,7 +72,13 @@ loaded by the Makefile (via `-include .env`) to inject environment variables
 when you run `make` targets:
 
 ```shell
+# Set your provider (supported: openAI, anthropic, azureOpenAI, gemini, ollama)
+KAGENT_DEFAULT_MODEL_PROVIDER=openAI
+
+# Set the corresponding API key for your provider
 OPENAI_API_KEY=your-openai-api-key
+# ANTHROPIC_API_KEY=your-anthropic-api-key
+# GOOGLE_API_KEY=your-google-api-key
 ```
 
 1. Build images, load them into kind cluster and deploy everything using Helm:

